packages/thirdweb/src/utils/any-evm/zksync/isZkSyncChain.ts

```diff
@@ -1,12 +1,38 @@
 import type { Chain } from "../../../chains/types.js";
+import { withCache } from "../../promise/withCache.js";
 
-export function isZkSyncChain(chain: Chain) {
-  return (
-    chain.id === 324 ||
-    chain.id === 300 ||
-    chain.id === 302 ||
-    chain.id === 11124 ||
-    chain.id === 282 || // cronos zkevm testnet
-    chain.id === 388 // cronos zkevm mainnet
+export async function isZkSyncChain(chain: Chain) {
+  const stack = await getChainStack(chain.id).catch(() => {
+    // fall back to checking against these zksync chain-ids
+    if (
+      chain.id === 324 ||
+      chain.id === 300 ||
+      chain.id === 302 ||
+      chain.id === 11124 ||
+      chain.id === 282 || // cronos zkevm testnet
+      chain.id === 388 // cronos zkevm mainnet
+    ) {
+      return "zksync-stack";
+    }
+
+    return "";
+  });
+
+  return stack === "zksync-stack";
+}
+
+async function getChainStack(chainId: number): Promise<string> {
+  return withCache(
+    async () => {
+      const res = await fetch(`https://${chainId}.rpc.thirdweb.com/stack`);
+      const data = await res.json();
+
+      if (!res.ok) {
```

Comment:
Would even remove this. Just let it throw without any checks

Reply:
eh, even errors CAN parse valid JSON meaning it would trust the error

it's probably good to check this BUT:
1. it should happen before we do `res.json()`
2. it should also call `res.body?.cancel()` so that it does not leak memory